sdk/core/core-http/src/policies/proxyPolicy.ts

```diff
@@ -30,43 +29,48 @@ function loadEnvironmentProxyValue(): string | undefined {
   return httpsProxy || allProxy || httpProxy;
 }
 
-// Check whether the given `uri` matches the noProxyList. If it matches, any request sent to that same `uri` won't set the proxy settings.
+// Check whether the host of a given `uri` is in the noProxyList.
+// If there's a match, any request sent to the same host won't have the proxy settings set.
 function isBypassed(uri: string): boolean | undefined {
-  if (byPassedList.has(uri)) {
-    return byPassedList.get(uri);
+  if (noProxyList.length === 0) {
+    return false;
   }
-  loadNoProxy();
-  let isBypassedFlag = false;
   const host = URLBuilder.parse(uri).getHost()!;
-  for (const proxyString of noProxyList) {
-    if (proxyString[0] === ".") {
-      if (uri.endsWith(proxyString)) {
+  if (byPassedList.has(host)) {
+    return byPassedList.get(host);
+  }
+  let isBypassedFlag = false;
+  for (const pattern of noProxyList) {
+    if (pattern[0] === ".") {
```

Comment:
Am I allowed to set a NO_PROXY as `*.foo.com` directly?

Reply:
In this implementation the pattern will not match any hosts. As mentioned above, NO_PROXY is not a standard.  Some implementation only allows a single `*` to by pass all hosts, while some allow `*.foo.com`.  I logged #13391 for further discussion of NO_PROXY support.